File: 10. 이외 알고리즘/12/24 Basic Review/1283.py
n=int(input())
alpha=[0]*26
for i in range(n):
    s=input()
    idx=0;
    for j in range(len(s)):
        c=int(s[j].lower())-int('a')
        if(alpha[c]==0):
            alpha[c]=1
            idx=j
    for j in range(len(s)):
        if(j==idx):
            print(f"[%s]"%s[j],end="")
            continue
        print(s[j],end="")

File: 6. 구현(Simulation)/1283.cpp
#include<bits/stdc++.h>
#define fastio ios_base::sync_with_stdio(0),cin.tie(0),cout.tie(0)
using namespace std;

bool alpha[27];
int main(){
	fastio;
	int n;cin>>n;
	string s;
	for(int i=0;i<n;i++){
		getline(cin,s); 
		int idx;
		for(int j=0;j<s.length();j++){
			if(s[j]==' ') continue;
			if(!alpha[tolower(s[j])-'a']){
				idx=j;
				break;
			}
		}
		for(int j=0;j<s.length();j++){
			if(j==idx){
				cout<<"["<<s[j]<<"]";
				continue;
			}
			cout<<s[j];
		}
		cout<<"\n";
	}
}
